sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactService.java

```diff
@@ -1373,9 +1383,81 @@ public void delete(Contact contact) {
 
 		// Notify external journal if necessary
 		externalJournalService.handleExternalJournalPersonUpdateAsync(contact.getPerson().toReference());
+		deleteContactLinks(contact);
+
 		super.delete(contact);
 	}
 
+	@Override
+	public void deletePermanent(Contact contact) {
+		// Delete all tasks associated with this contact
+		List<Task> tasks = taskService.findBy(new TaskCriteria().contact(new ContactReferenceDto(contact.getUuid())), true);
```

Comment:
I believe you can remove the task deletion from the `deleteContact` method above because, afaik, it only needs to be done once we permanently delete the contact (since tasks are not shown if they are associated to deleted entities in the first place). We did the same for cases.

Reply:
updated